### PR TITLE
Don't Email For Receipts on Salary Accounts

### PR DESCRIPTION
**Pull request by sampoder**
I have a feeling like this might be too niche so feel free to reject, it would be kinda cool

### CODE REVIEW

**Review by garyhtou — APPROVED** [review]
💸

File: app/views/canonical_pending_transaction_mailer/notify_approved.html.erb [hunk 1]
```diff
@@ -1,7 +1,9 @@
 A charge of <%= render_money @cpt.amount_cents %> at '<%= @cpt.memo %>' occurred on your card ending in <%= @cpt.stripe_card.last4 %>.
 
-<%= link_to "Upload your receipt for this charge. (this link can used by anyone for two weeks)", @upload_url %>
+<% if !@event.salary %>
```
Comment (sampoder) [inline app/views/canonical_pending_transaction_mailer/notify_approved.html.erb]:
```suggestion
<% if @cpt.local_hcb_code.receipt_required? %>
```

File: app/mailers/canonical_pending_transaction_mailer.rb [hunk 2]
```diff
@@ -4,14 +4,15 @@ class CanonicalPendingTransactionMailer < ApplicationMailer
...
+    @event = @cpt.event
     @receipt_upload_feature = Flipper.enabled?(:receipt_email_upload_2022_05_10, @cpt.stripe_card.user)
     @upload_url = Rails.application.routes.url_helpers.attach_receipt_hcb_code_url(
       id: @cpt.local_hcb_code.hashid,
       s: @cpt.local_hcb_code.signed_id(expires_in: 2.weeks, purpose: :receipt_upload)
     )
 
     to = @cpt.stripe_card.user.email
-    subject = "Upload a receipt for your transaction at #{@cpt.smart_memo}"
+    subject = "#{@event.salary ? "New transaction" : "Upload a receipt for your transaction"} at #{@cpt.smart_memo}"
```
Comment (sampoder) [inline app/mailers/canonical_pending_transaction_mailer.rb]:
```suggestion
    subject = "#{@cpt.local_hcb_code.receipt_required? ? "New transaction" : "Upload a receipt for your transaction"} at #{@cpt.smart_memo}"
```

File: app/mailers/canonical_pending_transaction_mailer.rb [hunk 3]
```diff
@@ -4,14 +4,15 @@ class CanonicalPendingTransactionMailer < ApplicationMailer
   def notify_approved
     @cpt = CanonicalPendingTransaction.find(params[:canonical_pending_transaction_id])
     @user = @cpt.stripe_card.user
+    @event = @cpt.event
```
Comment (sampoder) [inline app/mailers/canonical_pending_transaction_mailer.rb]:
```suggestion
```

**Pull request merged**